boa_engine/src/object/mod.rs

```diff
@@ -1371,7 +1433,7 @@ impl<'context> FunctionBuilder<'context> {
             } => {
                 *constructor = yes;
             }
-            Function::VmOrdinary { .. } => unreachable!("function must be native or closure"),
+            _ => unreachable!("function must be native or closure"),
```

Comment:
Why this change? it's usually more visual if you see all the options, and clippy warns against this if only one is missing.